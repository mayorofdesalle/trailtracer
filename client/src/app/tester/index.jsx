import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
