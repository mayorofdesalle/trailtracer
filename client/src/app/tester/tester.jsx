import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };