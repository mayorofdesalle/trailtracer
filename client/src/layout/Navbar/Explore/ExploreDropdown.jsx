import { forwardRef } from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';

import Dropdown from '@components/ui/Dropdown';
import Text from '@components/ui/Text';

const ExploreDropdownInner = styled(Dropdown)`
    height: 200%;
    width: 100%;
    top: clamp(2.25rem, min(5dvw, 5dvh), 5.5rem);
    right: 0;
    align-items: start;
`;

/**
 * ExploreDropdown
 * @description
 * Dropdown for exploring the website.
 **/
const ExploreDropdown = forwardRef(function ExploreDropdown(props, ref) {
    const { t } = useTranslation();

    return (
        <ExploreDropdownInner ref={ref} {...props}>
            <Text $heading $ratio={0.5}>{t('forms.explore')}</Text>
        </ExploreDropdownInner>
    );
});

export default ExploreDropdown;