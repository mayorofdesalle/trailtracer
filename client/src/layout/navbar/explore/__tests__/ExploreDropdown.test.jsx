import { describe, expect, it, render, screen, vi } from '@app/tester';

import ExploreDropdown from '../ExploreDropdown';

describe('ExploreDropdown', () => {

    it('should render correctly', () => {
        render(<ExploreDropdown data-testid='explore-dropdown' />);
        const dropdown = screen.getByTestId('explore-dropdown');
        expect(dropdown).toBeInTheDocument();
        expect(dropdown.firstChild).toBeInTheDocument();
        expect(dropdown.firstChild).toHaveTextContent(/explore/i);
    });

    it('should match snapshot', () => {
        const { asFragment } = render(<ExploreDropdown />);
        expect(asFragment()).toMatchSnapshot();
    });

});