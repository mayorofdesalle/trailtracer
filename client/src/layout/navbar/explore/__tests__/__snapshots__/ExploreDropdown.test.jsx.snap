// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExploreDropdown > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-dmyCSP sc-hLQSwg sc-kFCroH kHwwsn fkKYEp lgqhfG bcZcDb"
    tabindex="0"
  >
    <div
      class="sc-gLLuof dcSHkq"
    >
      forms.explore
    </div>
  </div>
</DocumentFragment>
`;
